src/bech32.cpp

```diff
@@ -158,14 +197,14 @@ std::string Encode(const std::string& hrp, const data& values) {
     return ret;
 }
 
-/** Decode a Bech32 string. */
-std::pair<std::string, data> Decode(const std::string& str) {
+/** Decode a Bech32 or Bech32m string. */
+DecodeResult Decode(const std::string& str) {
     bool lower = false, upper = false;
     for (size_t i = 0; i < str.size(); ++i) {
         unsigned char c = str[i];
-        if (c < 33 || c > 126) return {};
         if (c >= 'a' && c <= 'z') lower = true;
-        if (c >= 'A' && c <= 'Z') upper = true;
+        else if (c >= 'A' && c <= 'Z') upper = true;
+        else if (c < 33 || c > 126) return {};
     }
     if (lower && upper) return {};
```

Comment:
We don't currently expose any way to decode a TEX address from zcashd, but if we did, it would be necessary to check that all-uppercase addresses are accepted (but not generated). The test vectors don't have any examples of all-uppercase addresses. Note that they may occur in practice because an all-uppercase alphanumeric string is more efficiently represented in a QR code.